validator-auction-rest-server/lib/EthEventsClient.js

```diff
@@ -0,0 +1,169 @@
+            '_source': 'timestamp'
+        }, this._axisConfig)
+        if(response.data.hits.total > 0) {
+            return response.data.hits.hits[0]._source.timestamp
+        }
+        else {
+            throw new Error('Could not get block time')
+        }
+    }
+
+    async getAuctionStartInSeconds() {
+        const response = await axios.post(`${this._baseUrl}/event/search/`, {
+            'size': 1,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'term': {
+                                'address.raw': this._contractAddress
+                            }
+                        },
+                        {
+                            'term': {
+                                'event.raw': 'AuctionStarted'
+                            }
+                        }
+                    ]
+                }
+            },
+            '_source': 'args'
+        }, this._axisConfig)
+        return response.data.hits.total > 0 ? response.data.hits.hits[0]._source.args[0]['value.num'] : 0
+    }
+
+    async getBidEvents() {
+        const response = await axios.post(`${this._baseUrl}/event/search/`, {
+            'size': 1000,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'term': {
+                                'address.raw': this._contractAddress
+                            }
+                        },
+                        {
+                            'term': {
+                                'event.raw': 'BidSubmitted'
+                            }
+                        }
+                    ]
+                }
+            },
+            '_source': 'args'
+        }, this._axisConfig)
+
+        return response.data.hits.hits.map(hit => {
+            // better check if they are in that sequence
+            const firstArg = hit._source.args.find(a => a.pos === 0)
+            const secondArg = hit._source.args.find(a => a.pos === 1)
+            const thirdArg = hit._source.args.find(a => a.pos === 2)
+            return {
+                bidder: firstArg['value.hex'],
+                bidValue: secondArg['value.hex'],
+                timestamp: thirdArg['value.num']
+            }
+        }).sort((a, b) => Number.parseInt(a.timestamp) - Number.parseInt(b.timestamp))
+    }
+
+    async getWhitelistedAddresses() {
+        const response = await axios.post(`${this._baseUrl}/event/search/`, {
+            'size': 1000,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'term': {
+                                'address.raw': this._contractAddress
+                            }
+                        },
+                        {
+                            'term': {
+                                // TODO: Ask come for name
```

Comment:
The event will look like this:
`AddedToWhitelist(address[] whitelistedAddresses)`

Reply:
or actually I like whitelisted, so lets do
`AddressWhitelisted(address whitelistedAddress)`